lib/addons/p5.dom.js

```diff
@@ -784,6 +784,81 @@
     return self;
   };
 
+  /**
+   * Creates a colorPicker element in the DOM for color input.
+   * The .value() method will return a hex string (#rrggbb) of the color.
+   * The .color() method will return a p5.Color object with the current chosen color.
+   *
+   * @method createColorPicker
+   * @param {String|p5.Color} [value] default color of element
+   * @return {p5.Element} pointer to <a href="#/p5.Element">p5.Element</a> holding created node
+   * @example
+   * <div>
+   * <code>
+   * var inp1, inp2;
+   * function setup() {
+   *   createCanvas(100, 100);
+   *   background('grey');
+   *   var c1 = '#ff0000';
+   *   inp1 = createColorPicker(c1);
+   *   var c2 = color('yellow');
+   *   inp2 = createColorPicker(c2);
+   *   inp1.input(setShade1);
+   *   inp2.input(setShade2);
+   * }
+   *
+   * function draw() {
+   *   // Finding a shade between the two
+   *   var commonShade = lerpColor(inp1.color(), inp2.color(), 0.5);
+   *   fill(commonShade);
+   *   rect(20, 20, 60, 60);
+   * }
+   *
+   * function setShade1() {
+   *   console.log('You are choosing shade 1 to be : ', this.value());
+   * }
+   * function setShade2() {
+   *   console.log('You are choosing shade 2 to be : ', this.value());
+   * }
+   * </code>
+   * </div>
+   */
+  p5.prototype.createColorPicker = function(value) {
+    p5._validateParameters('createColorPicker', arguments);
+    var elt = document.createElement('input');
+    var self;
+    elt.type = 'color';
+    if (value) {
+      if (value instanceof p5.Color) {
+        elt.value = value.toString('#rrggbb');
+      } else {
+        p5.prototype._colorMode = 'rgb';
+        p5.prototype._colorMaxes = {
+          rgb: [255, 255, 255, 255],
+          hsb: [360, 100, 100, 1],
+          hsl: [360, 100, 100, 1]
+        };
+        var c = p5.prototype.color(value);
+        elt.value = c.toString('#rrggbb');
+      }
+    } else {
+      elt.value = '#000000';
+    }
+    self = addElement(elt, this);
+    // Method to return a p5.Color object for the given color.
+    self.color = function() {
+      if (value.mode) {
+        p5.prototype._colorMode = value.mode;
+      }
+      if (value.maxes) {
+        p5.prototype._colorMaxes = value.maxes;
+      }
+      var c = p5.prototype.color(this.elt.value);
```

Comment:
is this intermediary variable required?

Reply:
nope, we can remove them [LN841](https://github.com/processing/p5.js/pull/3089/files#diff-84dfcaf5d56a24f4ee13a32c644817b0R841), [LN856](https://github.com/processing/p5.js/pull/3089/files#diff-84dfcaf5d56a24f4ee13a32c644817b0R856) . Thank you, @TanviKumar !